### DISCUSSION TITLE
Native Android with RN mixed in?

### DISCUSSION THREAD

**Discussion by daveyleo** [wix/Detox#2976] [Q&A] [answered]
We are looking into merging our RN app that uses Detox for testing with a native app, I don't see any recent updates on getting Detox working in Native android.  We can't seem to get detox to work with Native app, it fails in launchApp().  

detox[50130] TRACE: [WSS_GET_FROM, #62559] {"messageId":-10000,"type":"AppWillTerminateWithError","params":{"errorDetails":"@Thread detox.primary(455):\njava.lang.ClassCastException: android.app.Application cannot be cast to com.facebook.react.ReactApplication\n\tat com.wix.detox.reactnative.ReactNativeExtension.waitForRNBootstrap(ReactNativeExtension.kt:57)\n\tat com.wix.detox.DetoxMain.initReactNativeIfNeeded(DetoxMain.kt:102)\n\tat com.wix.detox.DetoxMain.doInit(DetoxMain.kt:34)\n\tat com.wix.detox.DetoxMain.access$doInit(DetoxMain.kt:15)\n\tat com.wix.detox.DetoxMain$initActionHandlers$$inlined$with$lambda$1.handle(DetoxMain.kt:52)\n\tat com.wix.detox.adapters.server.ActionsExecutor$executeAction$$inlined$let$lambda$1.run(DetoxActionsDispatcher.kt:64)\n\tat android.os.Handler.handleCallback(Handler.java:883)\n\tat android.os.Handler.dispatchMessage(Handler.java:100)\n\tat android.os.Looper.loop(Looper.java:214)\n\tat com.wix.detox.adapters.server.ActionsExecutor$1.run(DetoxActionsDispatcher.kt:50)\n\tat java.lang.Thread.run(Thread.java:919)\n\nCheck device logs for full details!"}}
detox[50130] WARN:  [DetoxConnection.js/ERROR, #62559] Action dispatched too early, there is no session to use:

HINT: Please report this issue on our GitHub tracker:
https://github.com/wix/Detox/issues

{
  messageId: -10000,
  type: 'AppWillTerminateWithError',
  params: [Object]
}

seems  to want to be a RN app : "android.app.Application cannot be cast to com.facebook.react.ReactApplication"

digging into the code it seems to check for React with clsss.forName and then blindly cast the application.  So the way its coded it's all or nothing?

So it isn't possible to run a "hybrid" native + react-native app with detox tests?   

If true, what a bummer I tried this same thing with IOS and it works. :(

**Comment by d4vidi** [comment, marked as answer]
We do not currently support pure native / RN-brownfield Android apps with Detox. This is a long overdue request coming from the community every now and then. We might consider addressing it in the next quarter. ATM, we don't have the resources.

**Reply by daveyleo** [reply]
Thanks for the reply, I'll keep watching for updates, would really be nice if Android Native support was added in the near future, as our project isn't merging right away but will be in the next 6 months or so.
